src/expressions/dataTypes/atomize.ts

```diff
@@ -31,50 +42,58 @@ export function atomizeSingleValue(
 	const domFacade = executionParameters.domFacade;
 
 	if (isSubtypeOf(value.type, 'node()')) {
-		const /** Pointer */ pointer = value.value;
+		const pointer = value.value;
 
 		// TODO: Mix in types, by default get string value.
 		// Attributes should return their value.
 		// Text nodes and documents should return their text, as untyped atomic
-		if (isSubtypeOf(value.type, 'attribute()') || isSubtypeOf(value.type, 'text()')) {
+		if (
+			pointer.node.nodeType === NODE_TYPES.ATTRIBUTE_NODE ||
+			pointer.node.nodeType === NODE_TYPES.TEXT_NODE
+		) {
 			return sequenceFactory.create(
-				createAtomicValue(domFacade.getData(pointer), 'xs:untypedAtomic')
+				createAtomicValue(domFacade.getDataFromPointer(pointer), 'xs:untypedAtomic')
 			);
 		}
 
 		// comments and PIs are string
 		if (
-			isSubtypeOf(value.type, 'comment()') ||
-			isSubtypeOf(value.type, 'processing-instruction()')
+			pointer.node.nodeType === NODE_TYPES.COMMENT_NODE ||
```

Comment:
Is it no longer possible to have a custom type which inherits comments or processing instructions?
@DrRataplan I'm not very familiar with the custom types do you know if we can safely substitute with `isSubtypeOf` with an `===`?
Same goes for the other places in this function, several usages of `isSubtypeOf` seem to have been replaced.

Reply:
in practice, the dom types are not subclassed. If we add support for that, we should go over all of the code; this just saves a bit of performance